kubectl-plugin/src/resources/utils.rs

```diff
@@ -0,0 +1,87 @@
+use prettytable::{format, Row, Table};
+use serde::ser;
+
+// Constant to specify the output formats, these should work irrespective of case.
+pub const YAML_FORMAT: &str = "yaml";
+pub const JSON_FORMAT: &str = "json";
+
+// Constants to store the table headers of the Tabular output formats.
+lazy_static! {
+    pub static ref VOLUME_HEADERS: Row =
+        row!["ID", "PATHS", "REPLICAS", "PROTOCOL", "STATUS", "SIZE"];
+    pub static ref POOLS_HEADERS: Row = row![
+        "ID",
+        "TOTAL CAPACITY",
+        "USED CAPACITY",
+        "DISKS",
+        "NODE",
+        "STATUS"
+    ];
+}
+
+// table_printer takes the above defined headers and the rows created at execution,
+// to create a Tabular output and prints to the stdout.
+pub fn table_printer(titles: Row, rows: Vec<Row>) {
+    let mut table = Table::new();
+    // FORMAT_CLEAN has been set to remove table borders
+    table.set_format(*format::consts::FORMAT_CLEAN);
+    table.set_titles(titles);
+    for row in rows {
+        table.add_row(row);
+    }
+    table.printstd();
+}
+
+// CreateRows trait to be implemented by Vec<Volume/Pool> to create the rows.
+pub trait CreateRows {
+    fn create_rows(&self) -> Vec<Row>;
+}
+
+// GetHeaderRow trait to be implemented by Volume/Pool to fetch the corresponding headers.
+pub trait GetHeaderRow {
+    fn get_header_row(&self) -> Row;
+}
+
+// OutputFormat to be used as an enum to match the output from args.
+#[derive(Debug)]
+pub enum OutputFormat {
+    Yaml,
+    Json,
+    NoFormat,
+}
+
+impl From<&str> for OutputFormat {
```

Comment:
btw you can implement TryFrom instead of From, if you want to use only "none" for NoFormat and make other things be an error